### ISSUE TITLE
Simplify sample creation form and allow setting final laboratory result without tests [3]

### ISSUE THREAD

**Issue by MateStrysewskeSym** [hzi-braunschweig/SORMAS-Project#1638]
### Situation Description
In the current COVID situation, people often want to add test results to cases without having any information about the sample used or details about the test.

### Feature Description
Add a new "test result" field to cases. This could have the values "Not yet tested", "Negative", "Positive" and "Unknown". Make sure to give the field a caption that clarifies that this is meant to reflect the testing status for the disease of the case. 

For now, these fields don't have to influence case classification or anything else and should only be enabled for Germany. Also, ask users to convert a contact to a case if it has been tested positive.

### Review
@MartinWahnschaffeSymeda

**Comment by MartinWahnschaffeSymeda** [comment]
Is highly related to #1715

**Comment by bernardsilenou** [comment]
@MateStrysewskeSym I suggest that we enable this test field for contacts for all users. Ghana Fiji and  Nigeria also test contact person co COVID

**Comment by MateStrysewskeSym** [comment]
@bernardsilenou I agree

**Comment by markusmann-vg** [comment]
do we need further refinement before we can take it into account?

**Comment by MateStrysewskeSym** [comment]
For contacts, likely not. I've already created an issue for that.

**Comment by MartinWahnschaffeSymeda** [comment]
Just adding a new field for the test result may not be sufficient. We probably need at least the type of sample (e.g. nasal swab). An alternative is to use the existing samples.

1. Simplify the sample creation form (e.g. hide sent/received detail fields when not needed, pre-select purpose as external)
2. Allow to set the final lab result of the sample without the need to create a dedicated sample test

The mockup show what this could look like:
https://app.moqups.com/3NUENEwuOE/view/page/a69b80bd9

![grafik](https://user-images.githubusercontent.com/23217632/78798111-e455e800-79b8-11ea-9e3b-4acd6b9bb9c1.png)

Note: Date sample was collected, type of sample and laboratory are still mandatory fields!
What is missing in this scenario is the type of sample test conducted (e.g. PCR). I'm not sure if this is needed.

**Comment by bernardsilenou** [comment]
Sorry to reply late. comments below:

1, Add field for “lab test manufacturer” as free text. This field will capture the manufacturer of the chemicals used to to the test. We should also add this field tot eh default test result interface.
2, Exchange “request pathogen tests to be performed” with a field to express “type of test performed”
3, Delete “sent/dispatched to lab” and “received at lab”
4, Add a field to choose disease which was tested for just as in the normal test interface  
5,  Add a field “ date and time sample was tested beside "date sample was collected"

**Comment by MateStrysewskeSym** [comment]
@bernardsilenou Adding most of the fields that you requested would lead to the samples essentially being pathogen tests. I don't think that makes sense, and I think we need to refine this in the next sprint review again.

**Comment by bernardsilenou** [comment]
OK, the goal is to have both sample and pathogen rest in one tab. Can we refine it before the next sprint? because this is urgent to be in the next release especially to German users

**Comment by markusmann-vg** [comment]
mostly cosmetic

=> Risk: final lab result can be defined without having a sample test in the system

**Comment by MartinWahnschaffeSymeda** [comment]
I'll create another mockup for this that includes the creation of a sample test.

**Comment by MartinWahnschaffeSymeda** [comment]
I have updated the mockups to include a better solution for the sample test result:

1. The user creates a new sample. There is a field for the test result in the form:
https://app.moqups.com/3NUENEwuOE/view/page/a69b80bd9

![grafik](https://user-images.githubusercontent.com/23217632/81692447-b3a51a80-945e-11ea-80ce-e4d79308f50d.png)

2. When the user enters a test result (optional), new fields appear to add the needed details for the sample test. When saving this, it will create a new sample AND a test result entry:
https://app.moqups.com/3NUENEwuOE/view/page/ad30b933a

![grafik](https://user-images.githubusercontent.com/23217632/81692954-7725ee80-945f-11ea-9158-f3d8d457b106.png)

**Comment by MartinWahnschaffeSymeda** [comment]
I have updated the mockups to include a better solution for the sample test result:

1. The user creates a new sample. There is a field for the test result in the form:
https://app.moqups.com/3NUENEwuOE/view/page/a69b80bd9

![grafik](https://user-images.githubusercontent.com/23217632/81692447-b3a51a80-945e-11ea-80ce-e4d79308f50d.png)

2. When the user enters a test result (optional), new fields appear to add the needed details for the sample test. When saving this, it will create a new sample AND a test result entry:
https://app.moqups.com/3NUENEwuOE/view/page/ad30b933a

![grafik](https://user-images.githubusercontent.com/23217632/81692954-7725ee80-945f-11ea-9158-f3d8d457b106.png)

@bernardsilenou @max-hzi Please have a look at this and give feedback. Do you think this approach of combined sample and test result entry would work in the field?

**Comment by MartinWahnschaffeSymeda** [comment]
Updated mockups to include a way to directly create a test result together with the sample.

1. User can create a new sample. The whole form would be a bit slimmer now and has the additional optional test result field:
https://app.moqups.com/3NUENEwuOE/view/page/a69b80bd9

![grafik](https://user-images.githubusercontent.com/23217632/81698815-cde2f680-9466-11ea-85a7-a9a83d462372.png)

2. When a test result is chosen the additional fields needed for the test result will show up:
https://app.moqups.com/3NUENEwuOE/view/page/ad30b933a

![grafik](https://user-images.githubusercontent.com/23217632/81698919-eeab4c00-9466-11ea-96ca-bec6089e7614.png)

@bernardsilenou @max-hzi Do you think this will work for the users in the field?

@Chinedar Do you think this would make sense for Nigeria as-well? Would it be a problem if supervisors that create a sample in the web app can also directly create the sample test for it?

**Comment by max-hzi** [comment]
@bernardsilenou May I ask you to comment on this? I'm not sure if I am able to judge. But from what I can say it looks good. Thank you.

**Comment by MartinWahnschaffeSymeda** [comment]
Clarification: This should only be used when creating samples, not when editing them.
Also the app part of this is not part of the scope. If needed when can add it to the app later.

**Comment by bernardsilenou** [comment]
So sorry i missed this issue, just seeing it now.
I also agree and this is is very explicit to use.
@MartinWahnschaffeSymeda  Changes after discussing with team members:

1. Remove purpose of sample, tag them at back end as external by default and maybe display it to the user.
2. Remove request test, dispatch and received
3, Remove result verified by supervisor

**Comment by MateStrysewskeSym** [comment]
@bernardsilenou "Result verified by supervisor" is a required field, we can't get rid of that.

What's the reason for removing the other fields?

**Comment by bernardsilenou** [comment]
The German users in 99,9% of the times record results from external testing. Their goal is to have a way of documenting results with the minimum variable. So that is why we decided to delete variables that are linked to the management process of the sample.
For "Result verified by supervisor", we can hide them  from German users and check it to yes by default

**Comment by MateStrysewskeSym** [comment]
@bernardsilenou Please note that this feature will be available for all users of all countries. We definitely shouldn't make changes here that only reflect the German needs.

**Comment by bernardsilenou** [comment]
I understand, Can you call me please?

**Comment by bernardsilenou** [comment]
@MateStrysewskeSym,  I called to no avail.
1, I think for external users, we should limit their used of this module, eg only Lab user, national users and surveillance supervisors. If this option is open to all users, then no one will use the default lab  and result creation anymore. 
2. We also need to activate this feature for contacts and cases  right?

**Comment by MateStrysewskeSym** [comment]
@bernardsilenou The way this works is that an actual pathogen test is created. This form is basically a shortcut to create the first pathogen test, and it contains all required variables that you would need to fill in when creating a pathogen test the usual way. I don't see this as a problem.

I don't quite understand your second question. Samples are a part of cases and contacts, so of course this is enabled for both.

**Comment by bernardsilenou** [comment]
@

> @bernardsilenou The way this works is that an actual pathogen test is created. This form is basically a shortcut to create the first pathogen test, and it contains all required variables that you would need to fill in when creating a pathogen test the usual way. I don't see this as a problem.

We must not delete from the result method,  when i mean is if we can check them by default and hide it from the user.  Our goal is to limit the number of boxes that the user need to click on the record a sample results. If we can check and hid the default boxes them it will be easy fro the user also.
> I don't quite understand your second question. Samples are a part of cases and contacts, so of course this is enabled for both.

This is what i mean, thenks

**Comment by bernardsilenou** [comment]
@MateStrysewskeSym  Since this is a pathogen test result creation, why cant we activate automatic case classification using it? For the situation of COVID, many countries will instead use this module to register lab results for contacts and cases. 
Positive test result for cases will mean classification =  confirmed, while for contacts it will trigger the method to create a confirmed case from a contact. Whats ur idea pls

**Comment by MateStrysewskeSym** [comment]
@bernardsilenou We still haven't received any form of automatic classification criteria for COVID-19 from your side. As soon as that has changed, we can of course enable automatic case classification for this disease.

Concerning your point above, that's not very clear to me. If we automatically assume that results are verified by supervisors, we don't need the field in the first place, but since it was specifically requested from Nigeria, I don't think we can and should simply hide and pre-fill it, right?

**Comment by MartinWahnschaffeSymeda** [comment]
@bernardsilenou Do you want to hide "result verified by lab supervisor" only for Germany or for other countries as-well?

Can you create a new issue with details on which fields of the sample and sample test should be hidden for which countries?

**Comment by BarnaBartha** [comment]
Findings during the review:
[ ] Check null pointer in case of no region or district
[ ] Fix description of empty sample list in contacts (still refers to case)
[ ] Check pathogen test result details mapping as it does not get saved

**Comment by bernardsilenou** [comment]
I will say we just hide "result verified by sup" be default for all instances. If the need be then we will create another issue latter as Martin suggested
